sklearn/linear_model/_quantile.py

```diff
@@ -0,0 +1,203 @@
+        self : object
+            Returns self.
+        """
+
+        X, y = self._validate_data(X, y, accept_sparse=False,
+                                   y_numeric=True, multi_output=False)
+
+        sample_weight = _check_sample_weight(sample_weight, X)
+
+        X, y, X_offset, y_offset, X_scale = self._preprocess_data(
+            X, y, fit_intercept=self.fit_intercept, copy=self.copy_X,
+            sample_weight=sample_weight)
+
+        if self.quantile >= 1.0 or self.quantile <= 0.0:
+            raise ValueError(
+                "Quantile should be strictly between 0.0 and 1.0, got %f"
+                % self.quantile)
+
+        n_obs, n_slopes = X.shape
+        n_params = n_slopes
+
+        X_full = X
+        if self.fit_intercept:
+            n_params += 1
+            X_full = np.concatenate([np.ones([n_obs, 1]), X], axis=1)
+
+        # the linear programming formulation of quantile regression
+        # follows https://stats.stackexchange.com/questions/384909/
+        c_vector = np.concatenate([
+            np.ones(n_params * 2) * self.alpha,
+            sample_weight * self.quantile,
+            sample_weight * (1 - self.quantile),
+        ])
+        # do not penalize the intercept
+        if self.fit_intercept:
+            c_vector[0] = 0
+            c_vector[n_params] = 0
+
+        a_eq_matrix = np.concatenate([
+            X_full,
+            -X_full,
+            np.eye(n_obs),
+            -np.eye(n_obs),
+        ], axis=1)
+        b_eq_vector = y
+
+        method = self.solver
+        if method == 'auto':
+            if sp_version < parse_version('1.0.0'):
+                method = 'simplex'
+            elif sp_version < parse_version('1.6.0'):
+                method = 'interior-point'
+            else:
+                method = 'highs'
+
+        result = linprog(
+            c=c_vector,
+            A_eq=a_eq_matrix,
+            b_eq=b_eq_vector,
+            method=method,
+            options=self.solver_options,
+        )
+        solution = result.x
+        if not result.success:
+            warnings.warn(
+                'Linear programming for Quantile regression did not converge. '
+                'Status is {}'.format(result.status), ConvergenceWarning
+            )
+            if solution is np.nan:
+                solution = np.zeros(a_eq_matrix.shape[1])
+
+        params_pos = solution[:n_params]
+        params_neg = solution[n_params:2 * n_params]
+        params = params_pos - params_neg
+
+        self.n_iter_ = result.nit
+
+        self.coef_ = params[self.fit_intercept:]
+        # do not use self.set_intercept_, because it assumes intercept is zero
+        # if the data is normalized, which is false in this case
+        if self.fit_intercept:
+            self.coef_ = self.coef_ / X_scale
+            self.intercept_ = params[0] + y_offset \
+                - np.dot(X_offset, self.coef_.T)
+        else:
+            self.intercept_ = 0.0
```

Comment:
Depends on the decision/insight on `_preprocess_data` above. Maybe, this can be removed.